src/linux.rs

```diff
@@ -300,8 +300,17 @@ impl<'a> I2CTransfer<'a> for LinuxI2CDevice {
 
     /// Issue the provided sequence of I2C transactions
     fn transfer(&mut self, messages: &'a mut [Self::Message]) -> Result<u32, LinuxI2CError> {
+        let msg_type = |flag: u16| flag & I2CMessageFlags::READ.bits();
+        let mut prev_msg_type = !msg_type(messages[0].flags);
```

Comment:
Indexing the slice like this could potentially panic if an empty slice is passed as a parameter (whereas the existing code would return an error from the FFI call). It doesn't seem like there's any value in calling the FFI function at all if the slice is empty, so maybe just make this function exit early in that case?

Reply:
Yes, returning early is probably the safer option. but based on the other comment maybe we can get away with a `messages.first().map(|m| msg_type(m.flags))`?